src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -141,8 +145,29 @@ def _set_params(
             metrics_kwargs_ = _default_function_kwargs.copy()
             metrics_kwargs_.update(metrics_kwargs)
 
+        all_metric_names = metric_names
+        tm_extension = self.sorting_analyzer.get_extension("template_metrics")
+        if delete_existing_metrics is False and tm_extension is not None:
+            existing_metric_names = tm_extension.params["metric_names"]
+            existing_params = tm_extension.params["metrics_kwargs"]
+
+            # checks that existing metrics were calculated using the same params
+            if existing_params != metrics_kwargs_:
+                warnings.warn(
+                    "The parameters used to calculate the previous template metrics are different than those used now. Deleting previous template metrics..."
+                )
+                self.sorting_analyzer.get_extension("template_metrics").data["metrics"] = pd.DataFrame(
```

Comment:
Are this and the next three lines all functioning to completely delete existing metrics? If so this could be a candidate for a new function `self.delete_existing_metrics()`, just so if people want to do this in future there is no risk of forgetting one of these steps and half-deleting the metrics.

Reply:
I've simplified this, so that we don't need to reset everything here. Before, we needed non-empty results here, which were then appended to when we computed the new metrics. Now, the results dict is initialised at `_compute_metrics`, then the existing results are appended. So we don't need to initialise an empty DataFrame here.